source/DownloadHelper.cpp

```diff
@@ -0,0 +1,192 @@
+/* DownloadHelper.cpp
+Copyright (c) 2023 by RisingLeaf(https://github.com/RisingLeaf)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "DownloadHelper.h"
+
+#include <archive.h>
+#include <archive_entry.h>
+#include <cstring>
+#include <curl/curl.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <unistd.h>
```

Comment:
Pretty sure some of these includes won't work on Windows

Reply:
which ones?

Reply:
appearently unistd, fixed it though